libtiny_client/src/state.rs

```diff
@@ -512,6 +654,41 @@ impl StateInner {
             }
         }
     }
+
+    /// If channel is in Joining state cancel Joining task, otherwise sent part message
+    fn leave_channel(&mut self, msg_chan: &mut Sender<Cmd>, chan: &str) {
+        if let Some(idx) = utils::find_idx(&self.chans, |c| c.name == chan) {
+            match &mut self.chans[idx].join_state {
+                JoinState::Initial => {
+                    warn!("Unexpected state. Leaving channel before attempting to join.")
```

Comment:
Another unexpected state.

Reply:
I think this is not so unexpected. It happens when: 

- We already have a tab for a channel (e.g. from before the connection timed out)
- We re-connect to the server and send a JOIN message
- But we haven't heard back from the server yet (i.e. no JOIN or 477 response yet)
- User closes the tab

I think this is fine, no need for logging.